### PR TITLE
[MM-873]: Added server testcases for api/event.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added server testcases for api/event.go

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: calendar/api/events_test.go [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				requestBody := model.PostActionIntegrationRequest{
+					Context: map[string]interface{}{
+						"value": true,
+					},
+				}
+				bodyBytes, _ := json.Marshal(requestBody)
+				req.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+
+				mockStore.EXPECT().LoadUser(MockUserID).Return(nil, store.ErrNotFound)
```
Comment (raghavaggarwal2308) [inline calendar/api/events_test.go:247]: Can we add .Times() in all these mocks

File: calendar/api/events_test.go [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				currentTime := time.Now()
+				validJSON := GetCurrentTimeRequestBodyJSON()
+				req.Body = io.NopCloser(bytes.NewBufferString(validJSON))
+				mockOAauthToken := oauth2.Token{}
+				mockStore.EXPECT().LoadUser(MockUserID).Return(&store.User{MattermostUserID: MockUserID, OAuth2Token: &mockOAauthToken, Remote: &remote.User{ID: MockRemoteUserID}}, nil)
+				mockPluginAPI.EXPECT().CanLinkEventToChannel(MockChannelID, MockUserID).Return(true)
+				mockRemote.EXPECT().MakeClient(gomock.Any(), &mockOAauthToken).Return(mockRemoteClient)
+				mockRemoteClient.EXPECT().GetMailboxSettings(MockRemoteUserID).Return(&remote.MailboxSettings{TimeZone: "UTC"}, nil)
+				mockEvent := &remote.Event{
```
Comment (raghavaggarwal2308) [inline calendar/api/events_test.go:424]: We should create a test util for this as it is being created multiple times

Comment (raghavaggarwal2308) [reply calendar/api/events_test.go:424]: If not possible for the whole object we can divide it into chunks

File: calendar/api/events_test.go [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+		},
+		{
+			name: "Event created successfully without channelID",
+			setup: func(req *http.Request) {
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				currentTime := time.Now()
+				date := currentTime.Format("2006-01-02")
+				startTime := currentTime.Add(time.Hour).Format("15:04")
+				endTime := currentTime.Add(2 * time.Hour).Format("15:04")
+				validJSON := fmt.Sprintf(`{
```
Comment (raghavaggarwal2308) [inline calendar/api/events_test.go:553]: We have created a constant similar to this. Cant we use that here? Maybe we can create a util which accepts a param channeID

File: calendar/api/test_utils.go [hunk 4]
```diff
@@ -129,3 +142,37 @@ func GetMockSetup(t *testing.T) (*api, *mock_store.MockStore, *mock_bot.MockPost
...
+}
+
+func GetCurrentTimeRequestBodyJSON() string {
+	currentTime := time.Now()
+	date := currentTime.Format("2006-01-02")
+	startTime := currentTime.Add(time.Hour).Format("15:04")
+	endTime := currentTime.Add(2 * time.Hour).Format("15:04")
+
+	return fmt.Sprintf(`{
+					"all_day": false,
```
Comment (raghavaggarwal2308) [inline calendar/api/test_utils.go:168]: Same for this

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**